### PR TITLE
DAOS-8346 test: Use context manager for daos server in NLT.

### PR DESCRIPTION
**Pull request by ashleypittman**
Avoid a bug where large log files could be processed twice, leading
to increased runtime.  What was happening was the object was handling
the file in stop(), and again in __del__() if it still existed, and
because the bzip2 process was run in the background it might have been,
plus because of the way python handles objects there's very little
control over when __del__ is called.

Remove the logfile from the object once it's processed as a first-off
which will avoid the bug, but use DaosServer under a context manager
which should destroy the object in a more timely manner as well.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: utils/node_local_test.py [hunk 1]
```diff
@@ -440,10 +440,12 @@ def dfuse_mount_name(self):
 class DaosServer():
     """Manage a DAOS server instance"""
 
-    def __init__(self, conf, test_class=None, valgrind=False):
+    def __init__(self, conf, test_class=None, valgrind=False, wf=None, fe=None):
         self.running = False
         self._file = __file__.lstrip('./')
         self._sp = None
+        self.wf = wf
+        self.fe = fe
```
Comment (daltonbohning) [inline utils/node_local_test.py:448]: If you have to push again, could you please add either a docstring or inline comment for these params? Specifically, `wf` and `fe` aren't immediately intuitive (to me, at least). I think `we=WarningsFactory` and `fe=fatal_errors`?

**Review by daltonbohning — APPROVED** [review]
This cleans up nicely. For the pylint warnings, I wonder if we could put
`# pylint: disable=consider-using-with`
at the top of this file to remove all the warnings? Or at least on a per-line / per-usage basis:
`fs = open() # pylint: disable=consider-using-with`

File: utils/node_local_test.py (hunk 1 above)
Comment (ashleypittman) [reply utils/node_local_test.py:448]: For the pylint warnings I was planning to address them (in fact it's the warnings that made me aware that I could make the improvements in this PR).  I've not had time to do this though, and am wary of conflicts with so many changes in flight but I think you're right and we should just wholesale silence the noise to improve quality of reporting.

Yes to these variable names.  I'll make the suggested changes if I need to repush.

Comment (daltonbohning) [reply utils/node_local_test.py:448]: Makes sense, and thanks!

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**